src/Images/SampleDistanceSlider.jsx

```diff
@@ -1,66 +1,73 @@
 import React, { useEffect, useRef } from 'react'
-import { useActor } from '@xstate/react'
-import { Icon, Slider, Tooltip } from '@mui/material'
+import { useSelector } from '@xstate/react'
 import { sampleDistanceIconDataUri } from 'itk-viewer-icons'
 import applyContrastSensitiveStyleToElement from '../applyContrastSensitiveStyleToElement'
 import '../style.css'
+import Button from 'react-bootstrap/Button'
+import Form from 'react-bootstrap/Form'
+import Image from 'react-bootstrap/Image'
+import Tooltip from 'react-bootstrap/Tooltip'
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
+import cn from 'classnames'
 
-function SampleDistanceSlider(props) {
+const SampleDistanceSlider = React.memo(function SampleDistanceSlider(props) {
   const { service } = props
   const spacingDiv = useRef(null)
   const spacingElement = useRef(null)
-  const [state, send] = useActor(service)
-  const name = state.context.images.selectedName
-  const actorContext = state.context.images.actorContext.get(name)
+  const stateContext = useSelector(service, (state) => state.context)
```

Comment:
I spy with my little eye...

Reply:
Oh 💡 
I can use multiple selectors here as you suggested [here](https://github.com/InsightSoftwareConsortium/itk-viewer-bootstrap-ui/pull/7/files#r888410436) .